ast/env.go

```diff
@@ -107,14 +107,31 @@ func (env *TypeEnv) Get(x interface{}) types.Type {
 
 	// Comprehensions.
 	case *ArrayComprehension:
-
 		checker := newTypeChecker()
 		cpy, errs := checker.CheckBody(env, x.Body)
 		if len(errs) == 0 {
 			return types.NewArray(nil, cpy.Get(x.Term))
 		}
 
 		return nil
+	case *ObjectComprehension:
+		checker := newTypeChecker()
+		cpy, errs := checker.CheckBody(env, x.Body)
+		if len(errs) == 0 {
+			// TODO(mmussomele): How to set an object comprehension's
```

Comment:
I think comment is out-of-date. We need to come back to add key types for both objects and object comprehensions though.